src/test/java/seedu/taassist/logic/commands/DeletesCommandTest.java

```diff
@@ -0,0 +1,141 @@
+import static seedu.taassist.commons.core.Messages.MESSAGE_SESSION_DOES_NOT_EXIST;
+import static seedu.taassist.logic.commands.CommandTestUtil.VALID_SESSION_LAB1;
+import static seedu.taassist.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.taassist.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.taassist.testutil.Assert.assertThrows;
+import static seedu.taassist.testutil.TypicalSessions.ASSIGNMENT_1;
+import static seedu.taassist.testutil.TypicalSessions.LAB_1;
+import static seedu.taassist.testutil.TypicalSessions.TUTORIAL_1;
+import static seedu.taassist.testutil.TypicalStudents.getTypicalTaAssist;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import seedu.taassist.logic.commands.exceptions.CommandException;
+import seedu.taassist.model.Model;
+import seedu.taassist.model.ModelManager;
+import seedu.taassist.model.UserPrefs;
+import seedu.taassist.model.moduleclass.ModuleClass;
+import seedu.taassist.model.session.Session;
+import seedu.taassist.model.stubs.ModelStub;
+import seedu.taassist.model.stubs.ModelStubNeverInFocusMode;
+import seedu.taassist.testutil.ModuleClassBuilder;
+import seedu.taassist.testutil.SessionBuilder;
+
+public class DeletesCommandTest {
+
+    private final Model model = new ModelManager(getTypicalTaAssist(), new UserPrefs());
+    private final Model expectedModel = new ModelManager(model.getTaAssist(), new UserPrefs());
+
+    //==================================== Unit Tests ================================================================
+
+    @Test
+    public void constructor_nullSession_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new DeletesCommand(null));
+    }
+
+    @Test
+    public void execute_notInFocusMode_throwsCommandException() {
+        ModelStubNeverInFocusMode modelStub = new ModelStubNeverInFocusMode();
+        Session validSession = new SessionBuilder().withName(VALID_SESSION_LAB1).build();
+        Set<Session> validSessions = new HashSet<>(List.of(validSession));
+        assertThrows(CommandException.class, () -> new DeletesCommand(validSessions).execute(modelStub));
+    }
+
+    @Test
+    public void execute_existingSessionInFocusMode_success() throws CommandException {
+        ModelStubWithModuleClassWithSessions modelStub = new ModelStubWithModuleClassWithSessions();
+        assert !modelStub.getFocusedClass().getSessions().isEmpty();
+        Session existingSession = modelStub.getFocusedClass().getSessions().get(0);
+        Set<Session> existingSessions = new HashSet<>(List.of(existingSession));
+        DeletesCommand command = new DeletesCommand(existingSessions);
+        CommandResult commandResult = command.execute(modelStub);
+        assertEquals(DeletesCommand.getCommandMessage(existingSessions), commandResult.getFeedbackToUser());
```

Comment:
Don't we need to verify that `removeSessions` method was called with the correct sessions?

Similar issue in other tests bellow.